pages/06-higher-order-functions.md

```diff
@@ -64,111 +59,76 @@ to another function or to seed them with some data.
 look at this:
 
-~~~~ {.haskell:ghci name="code"}
-ghci> let multTwoWithNine = multThree 9
-ghci> multTwoWithNine 2 3
-54
-ghci> let multWithEighteen = multTwoWithNine 2
-ghci> multWithEighteen 10
-180
-~~~~
+```elm
+> multTwoWithNine = multThree 9
+<function> : number -> number -> number
+> multTwoWithNine 2 3
+54 : number
+> multWithEighteen = multTwoWithNine 2
+<function> : number -> number
+> multWithEighteen 10
+180 : number
+```
 
 By calling functions with too few parameters, so to speak, we're
 creating new functions on the fly. What if we wanted to create a
 function that takes a number and compares it to 100? We could do
 something like this:
 
-~~~~ {.haskell:hs name="code"}
-compareWithHundred :: (Num a, Ord a) => a -> Ordering
+```elm
+compareWithHundred : number -> Order
 compareWithHundred x = compare 100 x
-~~~~
+```
 
-If we call it with 99, it returns a GT. Simple stuff. Notice that the x
+If we call it with `99`, it returns a `GT`. Simple stuff. Notice that the `x`
 is on the right hand side on both sides of the equation. Now let's think
-about what compare 100 returns. It returns a function that takes a
-number and compares it with 100. Wow! Isn't that the function we wanted?
+about what `compare 100` returns. It returns a function that takes a
+`number` and compares it with `100`. Wow! Isn't that the function we wanted?
 We can rewrite this as:
 
-~~~~ {.haskell:hs name="code"}
-compareWithHundred :: (Num a, Ord a) => a -> Ordering
+```elm
+compareWithHundred : number -> Order
 compareWithHundred = compare 100
-~~~~
+```
 
-The type declaration stays the same, because compare 100 returns a
-function. Compare has a type of (Ord a) =\> a -\> (a -\> Ordering) and
-calling it with 100 returns a (Num a, Ord a) =\> a -\> Ordering. The
-additional class constraint sneaks up there because 100 is also part of
-the Num typeclass.
+The type declaration stays the same, because `compare 100` returns a
+function. Compare has a type of `number -> (number -> Order)` and
+calling it with `100` returns a `number -> Order`.
 
 *Yo!* Make sure you really understand how curried functions and partial
 application work because they're really important!
 
-Infix functions can also be partially applied by using sections. To
-section an infix function, simply surround it with parentheses and only
-supply a parameter on one side. That creates a function that takes one
-parameter and then applies it to the side that's missing an operand. An
-insultingly trivial function:
-
-~~~~ {.haskell:hs name="code"}
-divideByTen :: (Floating a) => a -> a
-divideByTen = (/10)
-~~~~
-
-Calling, say, divideByTen 200 is equivalent to doing 200 / 10, as is
-doing (/10) 200. A function that checks if a character supplied to it is
-an uppercase letter:
-
-~~~~ {.haskell:hs name="code"}
-isUpperAlphanum :: Char -> Bool
-isUpperAlphanum = (`elem` ['A'..'Z'])
-~~~~
-
-The only special thing about sections is using -. From the definition of
```

Comment:
As far as I know, none of this really applies to Elm. I also changed the `divideByTen` example above for the sake of simplicity, as it would have required using `flip`, which will be introduced later.